lib/widget/image_cropper.dart

```diff
@@ -0,0 +1,333 @@
+      'width': () => _controller.imageWidth,
+      'height': () => _controller.imageHeight,
+      'placeholderColor': () => _controller.placeholderColor,
+    };
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {
+      'crop': () => _controller.cropAction?.cropImage(),
+      'reset': () => _controller.cropAction?.reset(),
+      'rotateLeft': () => _controller.cropAction?.rotateLeft(),
+      'rotateRight': () => _controller.cropAction?.rotateRight(),
+    };
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'id': (value) => _controller.id = Utils.getString(value, fallback: ''),
+      'source': (value) =>
+          _controller.source = Utils.getString(value, fallback: ''),
+      'shape': (value) =>
+          _controller.shape = Utils.getString(value, fallback: 'Circle'),
+      'fit': (value) => _controller.fit = Utils.optionalString(value),
+      'width': (width) => _controller.imageWidth = Utils.optionalDouble(width),
+      'height': (height) =>
+          _controller.imageHeight = Utils.optionalDouble(height),
+      'borderRadius': (borderRadius) =>
+          _controller.shapeBorderRadius = Utils.optionalDouble(borderRadius),
+      'strokeColor': (value) => _controller.strokeColor = Utils.getColor(value),
+      'strokeWidth': (width) =>
+          _controller.strokeWidth = Utils.optionalDouble(width),
+      'cropPercentage': (cropPercentage) => _controller.cropPercentage =
+          Utils.optionalDouble(cropPercentage, max: 1.0),
+      'placeholderColor': (value) =>
+          _controller.placeholderColor = Utils.getColor(value),
+      'onTap': (funcDefinition) => _controller.onTap =
+          EnsembleAction.fromYaml(funcDefinition, initiator: this),
+      'isRotate': (value) =>
+          _controller.isRotate = Utils.getBool(value, fallback: true),
+      'isMove': (value) =>
+          _controller.isMove = Utils.getBool(value, fallback: true),
+      'isScale': (value) =>
+          _controller.isScale = Utils.getBool(value, fallback: true),
+      'onCropped': (funcDefinition) => _controller.onCropped =
+          EnsembleAction.fromYaml(funcDefinition, initiator: this)
+    };
+  }
+}
+
+mixin CropAction on WidgetState<EnsembleImageCropper> {
+  Future<void> cropImage();
+  void reset();
+  void rotateLeft();
+  void rotateRight();
+}
+
+class EnsembleImageCropperController extends BoxController {
+  CropAction? cropAction;
+  String source = '';
+  String? fit;
+  Color? placeholderColor;
+  EnsembleAction? onTap;
+  double? imageHeight;
+  double? imageWidth;
+
+  double? strokeWidth;
+  double? shapeBorderRadius;
+  double? cropPercentage;
+  String shape = '';
+  Color? strokeColor;
+  bool isRotate = true;
+  bool isMove = true;
+  bool isScale = true;
+  EnsembleAction? onCropped;
+  CustomImageCropController cropController = CustomImageCropController();
+}
+
+class EnsembleImageCropperState extends WidgetState<EnsembleImageCropper>
+    with CropAction {
```

Comment:
I don't understand this relationship - why does the state have all the logic? I think all the logic should be implemented on the controller and the state just fields in the values. @vusters WDYT?

Reply:
Can't the context be set on the controller during the build method? In any case, this seems good enough for now.